### PR TITLE
feat(AHWR-183): add enrich and send logic, steel thread

### PR DESCRIPTION
**Pull request by bendef-24**

### CODE REVIEW

**Review by omega1119 — APPROVED** [review]

File: app/config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import convict from 'convict'
-import convictFormatWithValidator from 'convict-format-with-validator'
+const convict = require('convict')
```
Comment (mattyod) [inline app/config.js:1]: shall we leave as [ES](https://defra.github.io/software-development-standards/standards/node_standards/#commonjs-vs-es-modules)?

File: app/index.js [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+const messaging = require('./messaging/index.js')
+const { createServer } = require('./server.js')
+
+let serverLogger
 
 const init = async () => {
   const server = await createServer()
   await server.start()
-  console.log('Server running on %s', server.info.uri)
+  serverLogger = server.logger
```
Comment (mattyod) [inline app/index.js:11]: Just as a general practice. Prefer not to rename things like this as it makes them harder to find in the codebase.

`server.logger.warn` is always `server.logger.warn` but `serverLogger.warn` only exists here. If we do feel we want to take something out of it's context, better to deconstruct? `{ logger } = server`;

Comment (bendef-24) [reply app/index.js:11]: Will rename to logger

File: app/messaging/message-request-schema.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const applicationMessageSchema = joi.object({
+  crn: joi.string().max(14).required(),
+  sbi: joi.string().max(14)
+})
+
+const validateMessageRequest = (logger, event) => {
+  const validate = applicationMessageSchema.validate(event)
+  if (validate.error) {
+    logger.error('Message request validation error', JSON.stringify(validate.error))
```
Comment (mattyod) [inline app/messaging/message-request-schema.js:11]: Another general point. Better to log the error as an object I think, Pino is heavily optimised for this and will catch errors during attempts to stringify. Also means we can consistently rely upon serialisers and redaction.

```
logger.error({  err }, 'message')
```

Comment (bendef-24) [reply app/messaging/message-request-schema.js:11]: Will update and commit once you've completed round one of code review. Thanks

File: app/config/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const Joi = require('joi')
+const dbConfig = require('./db')
+const messageQueueConfig = require('./message-queue')
+
+const schema = Joi.object({
+  port: Joi.number().default(3005),
+  env: Joi.string().valid('development', 'test', 'production').default('development'),
+  isDev: Joi.boolean().default(false),
+  termsAndConditionsUrl: Joi.string().default('#'),
```
Comment (aaroncarroll82) [inline app/config/index.js:9]: Are these url/uri actually required?

File: app/config/message-queue.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  useCredentialChain: process.env.NODE_ENV === 'production'
+}
+
+const config = {
+  sfdMessageRequestQueue: {
+    address: process.env.SFDMESSAGEREQUEST_QUEUE_ADDRESS,
+    type: 'queue',
+    ...sharedConfig
+  },
+  sfdMessageTopic: {
```
Comment (aaroncarroll82) [inline app/config/message-queue.js:39]: Is this the real topic, or the proxy/mock one? Maybe worth allowing both to be specified, and then having a toggle between the two.

File: app/index.js [hunk 6]
```diff
@@ -1,17 +1,34 @@
...
+  serverLogger.warn(`Server running on ${server.info.uri}`)
 }
 
 process.on('unhandledRejection', (err) => {
   console.log(err)
```
Comment (aaroncarroll82) [inline app/index.js:16]: Can this one not be a proper log entry too?

File: app/config.js (hunk 1 above)
Comment (mattyod) [reply app/config.js:1]: seeing as you did all the work in a single commit might be easier to just rebase that out?

File: app/messaging/forward-message-request-to-sfd.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+const sendSfdMessageRequest = async (sfdMessageRequest) => {
+  const sender = new MessageSender(sfdMessageTopic)
+  const message = createMessage(sfdMessageRequest, sfdMessageRequestMsgType)
+
+  await sender.sendMessage(message)
+  await sender.closeConnection()
+}
+
+const createMessage = (body, type) => {
```
Comment (aaroncarroll82) [inline app/messaging/forward-message-request-to-sfd.js:12]: I was expecting us to be matching the SFD contract as part of the scope of this ticket, but this is more of a dummy message

File: app/services/message-service.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+const buildSfdMessageFrom = (messageId, appMessage) => {
+  let data
+  if (appMessage.sbi) {
+    data = { ...appMessage }
+  }
+
+  return {
+    id: messageId,
+    agreementReference: appMessage.crn,
+    claimReference: 'fake-claim-1',
```
Comment (aaroncarroll82) [inline app/services/message-service.js:49]: ?

File: app/config.js (hunk 1 above)
Comment (bendef-24) [reply app/config.js:1]: Agreed, I did think about that but decided against it due to the tests failing to run, jest needs configuring to work with ES modules. I can do that in this PR if you feel strongly about it?

File: app/config/index.js (hunk 4 above)
Comment (aaroncarroll82) [reply app/config/index.js:9]: Resolving for now, hold over for future pr

File: app/config/message-queue.js (hunk 5 above)
Comment (aaroncarroll82) [reply app/config/message-queue.js:39]: Resolving for now, hold over for future pr

**Pull request merged**